src/shred_collector/shred_processor.zig

```diff
@@ -94,14 +108,32 @@ fn processShred(
     if (shred.* == .data) {
         const parent = try shred.data.parent();
         if (parent + 1 != slot) {
+            metrics.skipped_slot_count.add(slot - parent);
             tracker.skipSlots(parent, slot) catch |err| switch (err) {
-                error.SlotUnderflow, error.SlotOverflow => {},
+                error.SlotUnderflow, error.SlotOverflow => {
+                    metrics.skip_slots_error.observe(err);
+                },
             };
         }
     }
     if (shred.isLastInSlot()) {
         tracker.setLastShred(slot, index) catch |err| switch (err) {
-            error.SlotUnderflow, error.SlotOverflow => return,
+            error.SlotUnderflow, error.SlotOverflow => {
+                metrics.set_last_shred_error.observe(err);
+                return;
+            },
         };
     }
 }
+
+const Metrics = struct {
```

Comment:
My mentality behind the concise naming is that the context is already specified - in this case by the file name. This is the Metrics struct in shred_processor.zig, i.e. `shred_processor.Metrics`, i.e. the metrics for the shred processor. I'm fine calling this ShredProcessorMetrics if you think it improves clarity, it just seems unnecessary to me.

Regarding stats vs metrics I don't have a strong opinion on that either. Metrics seems like the natural choice since that's the terminology used in prometheus. But after I started that pattern I noticed that it conflicts with the Stats name in other places. So I'd definitely like to consolidate on a single name, so I'm happy to switch everything to Metrics

Maybe I'm being pedantic, but I think it's worth noting that the pattern as defined `{struct_name}Metrics` won't always apply, because metrics structs are not necessarily 1-to-1 with service structs. In the shred collector case, case there is no associated struct (except I guess the file that contains it, which is technically a struct). Sometimes there will be multiple associated structs, and sometimes there will be multiple metrics structs for one service struct.

Reply:
ok makes sense - im flexible, maybe we keep it `Metrics` if the context is already defined and theres no conflict

Reply:
Stats->Metrics: 90689ef28bc892cbea8d903212b7d3633e008e77